### ISSUE TITLE
[ENH] Pruning Methods for Causal Discovery 

### ISSUE THREAD

**Issue by Manas-7854** [pgmpy/pgmpy#3513]
**Is your feature request related to a problem? Please describe.**
A lot of causal discovery methods usually have a step for pruning of less important edges, we can have a class with a few of these pruning methods

**Describe the solution you'd like**
A class consisting of different pruning methods (CAM Pruning, PNS, etc)

**Additional context**
- For CAM Pruning, CAM scoring can be utilized - which is already an existing PR

For CAM Pruning:
    # Step 1: Initialize a copy of the adjacency matrix to store the pruned graph.
    # Step 2: Loop over each node (variable) in the graph.
    # Step 3: For each node, identify its current parents from the adjacency matrix.
    # Step 4: Use pgmpy's CAM score implementation (or a fallback OLS implementation) to score each parent.
    # Step 5: If the p-value of a parent's score exceeds `pruning_cutoff`,
    # remove the edge (set to 0 in the pruned adj matrix).
    # Step 6: Return the pruned adjacency matrix.
    raise NotImplementedError

For PNS:
    # Step 1: If `estimator` is None, instantiate the default `sklearn.ensemble.ExtraTreesRegressor(random_state=seed)`.
    # Step 2: Initialize a boolean mask of shape (d, d) with False.
    # Step 3: Loop over each variable `j` from 0 to d-1.
    # Step 4: For variable `j`, prepare target y = X[:, j] and features X_rest = X without column `j`.
    # Step 5: Fit the estimator on X_rest and y.
    # Step 6: Verify the fitted estimator has a `feature_importances_` attribute; raise TypeError if not.
    # Step 7: Calculate the importance threshold for variable `j` (pns_threshold * mean(feature_importances_)).
    # Step 8: Set the mask for column `j` (excluding the diagonal) to True for features with importance >= threshold.
    # Step 9: Return the boolean mask.
    raise NotImplementedError
